presto-main/src/main/java/com/facebook/presto/operator/SortedPositionLinks.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.Page;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkState;
+
+public final class SortedPositionLinks
+        implements PositionLinks
+{
+    private final PositionLinks positionLinks;
+    private final int[][] compactedPositionLinks;
+    private final Map<Integer, List<Integer>> positionLinksBuilder;
```

Comment:
I can not use `Multimap`, because I need to have last linked element as a starting point for the position links. Thus on each linking I would have to copy all collection returned by `Multimap.get(...)`